### DISCUSSION TITLE
Would MAUI User Interface designer for macOS rely on Xcode?

### DISCUSSION THREAD

**Discussion by McoreD** [dotnet/maui#217] [General]
As at today, Xcode is encouraged to graphically design User Interface in macOS apps: https://docs.microsoft.com/en-us/xamarin/mac/get-started/hello-mac

Would it be the case when MAUI is released, or would Visual Studio update to support UI building for macOS?

**Comment by mrwcjoughin** [comment]
The UI designer you are referring to is for Xamarin.macOS not Xamarin.Forms

MAUI is the evolution of Xamarin.Forms.
Xamarin.Forms already supports the building of UI's visually by using XAML.
Xamarin.Forms already supports XAML preview or XAML hot reload for iOS and
Android.
So I assume you are asking for this XAML preview or XAML hot reload to be
available when testing your XAML for macOS target?
If so yes that would be great, and I assume it is part of MAUI because
macOS is supposed to be a first-class target for MAUI

**Reply by McoreD** [reply]
You understood my question exactly although I didn't word out very well. Issue is currently we don't have a visual designer for macOS as target in Visual Studio. I hope it changes!

**Reply by jamesmcmenamin** [reply]
./vote for forgetting about macOS :P since macOS now supports iOS app's is there really a point in double-dipping... it's writing a uwp and a wpf both to target the same platform... not a good use of time :)

**Reply by Joebeazelman** [reply]
> ./vote for forgetting about macOS :P since macOS now supports iOS app's is there really a point in double-dipping... it's writing a uwp and a wpf both to target the same platform... not a good use of time :)

Why should MacOS get an inferior, least common denominator user experience? MacOS specific UIs are intended to run on desktop machines, while iOS specific ones are intended to run on mobile devices. Desktops UIs require a mouse, keyboard and a larger screen with layered Windows, while mobile devices UIs require small, touch screens with panels. Microsoft made a fatal mistake by creating a single UI for both formats and they're paying for it.

You clearly have an anti-mac agenda, but unbeknownst to you, a great deal of Windows, Linux and web developers use MacOS as their primary device. Even Microsoft's own internal developers use it.
